controller/appcontroller.go

```diff
@@ -46,6 +47,17 @@ const (
 	updateOperationStateTimeout = 1 * time.Second
 )
 
+type ComparisonLevel int
```

Comment:
I wonder about this name. How about `CompareWith` and then it would be `CompareWithLatest`, `CompareWithLastSync` and `CompareWithNothing`?

Reply:
Nice suggestion. Applied